AmahiAnywhere/AmahiAnywhere/Presentation/Login/LoginViewController.swift

```diff
@@ -41,27 +56,50 @@ class LoginViewController: BaseUIViewController {
         return true
     }
     
+    @IBAction func passwordFieldChanged(_ sender: Any) {
+        showHideButton.isHidden = !passwordInputField.hasText
+    }
+    
     @IBAction func userClickForgotPassword(_ sender: UIButton) {
         UIApplication.shared.open(NSURL(string:"https://www.amahi.org/forgot")! as URL)
     }
     
     @IBAction func userclickSignIn(_ sender: Any) {
+        if checkInputsValid(){
+            presenter.login(username: usernameInputField.text!,
+                            password: passwordInputField.text!)
+        }
+    }
+    
+    func checkInputsValid() -> Bool{
+        var isValid = true
         
         if (usernameInputField.text?.isEmpty)! {
-            usernameInputField.errorColor = UIColor.red
-            usernameInputField.errorMessage = StringLiterals.fieldIsRequired
-            return
+            showFieldErrorMessage(textField: usernameInputField)
+            isValid = false
         }
         
         if (passwordInputField.text?.isEmpty)! {
-            passwordInputField.errorColor = UIColor.red
-            passwordInputField.errorMessage = StringLiterals.fieldIsRequired
-            return
+            showFieldErrorMessage(textField: passwordInputField)
+            isValid = false
         }
         
-        presenter.login(username: usernameInputField.text!,
-                        password: passwordInputField.text!)
+        return isValid
+    }
+    
+    func showFieldErrorMessage(textField: SkyFloatingLabelTextField){
+        textField.errorColor = .red
+        textField.errorMessage = StringLiterals.fieldIsRequired
+    }
+    
+    @IBAction func userClickShowHide(_ sender: UIButton) {
+        if passwordInputField.isSecureTextEntry{
+            showHideButton.setTitle("Hide", for: .normal)
```

Comment:
Please move string literals to the StringLiteral file. I think an icon might be better for this instead of using text.

Reply:
Okay changed hide/show to an eye icon now just like how it looks on the android version. Is it all good now then? @codedentwickler